Add PHP 8.4 support

joanhey opened the pull request:
The tests pass, but show deprecations with the `symfony/string` v5.4.0 class in PHP 8.4 lowest test.

`Symfony/string` is fixed https://github.com/symfony/string/commit/bce75043af265dc8aca536a6ab1d6b3181763529 


~PD: perhaps it'll be better to change in `composer.json` :~
```
"require": {
        "php": "~8.2.0 || ~8.3.0 || ~8.4.0",
```
with
```
"require": {
        "php": ">=8.2",
```
 Changed to `"php": ">=8.2"`

Fix #871
that break PestPHP https://github.com/pestphp/pest/issues/1199

joanhey committed:
Add PHP ~8.4 in composer.json

joanhey committed:
Add PHP 8.4 to ci.yml workflow

joanhey committed:
Update composer.json with ~8.4.0

joanhey committed:
Update composer.json with PHP >=8.2

We always have time to limit before a new final PHP version.

Jean85 commented on composer.json:
```diff
@@ -33,7 +33,7 @@
     ],
     "require": {
-        "php": "~8.2.0 || ~8.3.0",
+        "php": ">=8.2",
```
Please do not put an open ended constraint here, the limited support to new PHP versions is intentional

joanhey committed:
Merge branch '7.x' into php8.4.x

joanhey committed:
Update composer.json

"php": "~8.2 || ~8.3 || ~8.4",

Constraints limited.
Used without the ".0" because have the same effect.

joanhey committed:
Update composer.json with patch version too

Slamdunk commented:
I don't like the CI build on 8.4 to be successful: https://github.com/paratestphp/paratest/actions/runs/10473712966/job/29006479710?pr=872#step:3:19

`pcov` is not yet compatible with 8.4, https://github.com/krakjoe/pcov/pull/111

joanhey committed:
Testing coverage with conditional

joanhey committed:
Try conditional again

joanhey committed:
Revert conditional

Composer.json require pcov.

joanhey committed:
Merge branch '7.x' into php8.4.x

Slamdunk committed:
Merge branch '7.x' into php8.4.x

Slamdunk committed:
Raise tests' timeout to 5 minutes

Slamdunk committed:
Raise tests' timeout to 5 minutes (ops)

Slamdunk commented:
Apparently there's an infinite loop somewhere with PHP 8.4, but I can't debug it yet locally because the pcov PECL extension hasn't been updated yet, we need to wait

Slamdunk commented:
Superseded by https://github.com/paratestphp/paratest/pull/898 thank you anyway for your contribution
